source/Lib/Utilities/ThreadPool.h

```diff
@@ -286,7 +286,46 @@ struct WaitCounter
   Barrier                         m_done{ false };
 };
 
+class PoolPause
+{
+public:
+  PoolPause() = default;
+  void setNrThreads(size_t nr) { m_nrThreads = nr; }
+  void unpauseIfPaused()
+  {
+    // All threads may be sleeping. If so, wake up.
+    std::unique_lock<std::mutex> lock(m_allThreadsWaitingMutex);
+    m_allThreadWaitingMoreWork = false;
+    m_allThreadsWaitingCV.notify_all();
+  }
+  void pauseIfAllOtherThreadsWaiting()
+  {
+    if (m_nrThreads == 0)
+      return;
+    auto nrWaiting = m_waitingForLockThreads.load( std::memory_order_relaxed );
+    if (nrWaiting == m_nrThreads - 1)
+    {
+      // All threads are waiting. This (current) threads is the one which locked `l`. All
+      // other threads are waiting in the above condition for `l.lock();`.
+      // The only way how more work for the threads can come in is if addBarrierTask is called 
+      // or if the thread pool is closed or destroyed.
+      std::unique_lock<std::mutex> lock(m_allThreadsWaitingMutex);
+      m_allThreadWaitingMoreWork = true;
+      while (m_allThreadWaitingMoreWork)
+      {
+        m_allThreadsWaitingCV.wait(lock);
+      }
+    }
+  }
+  
+  std::atomic_uint         m_waitingForLockThreads{ 0 };
```

Comment:
This could also just be a member of the threadPool. Also this stuff does not need to be in a class. I just thought this is a neat organization.